src/d/actor/d_a_npc_md.cpp

```diff
@@ -1629,41 +1629,64 @@ BOOL daNpc_Md_c::hitNpcAction(void* r29) {
 }
 
 /* 00008E54-00008FFC       .text setNormalSpeedF__10daNpc_Md_cFfffff */
-void daNpc_Md_c::setNormalSpeedF(f32 f1, f32 f2, f32 i_scale, f32 i_maxStep, f32 i_minStep) {
-    f32 f31 = mMaxNormalSpeed * f1;
-    int r3 = wallHitCheck();
-    if (r3 >= 0) {
-        s16 angle = current.angle.y + 0x8000 - mAcchCir[r3].GetWallAngleY();
+void daNpc_Md_c::setNormalSpeedF(f32 i_speedScale, f32 i_acceleration, f32 i_scale, f32 i_maxStep, f32 i_minStep) {
+    // Calculate the desired speed based on a scaling factor and the NPC's maximum normal speed
+    f32 desiredSpeed = mMaxNormalSpeed * i_speedScale;
+
+    // Check if Medli is colliding with a wall
+    int wallIndex = wallHitCheck();
```

Comment:
This returns the index of the AcchCir that is colliding with a wall, not the wall itself. Should be named "cir index" or something like that.